src/Sentry/SentryFailedRequestHandler.cs

```diff
@@ -0,0 +1,109 @@
+using Sentry.Protocol;
+
+namespace Sentry;
+
+internal class SentryFailedRequestHandler : ISentryFailedRequestHandler
+{
+    private readonly IHub _hub;
+    private readonly SentryOptions _options;
+
+    public const string MechanismType = "SentryFailedRequestHandler";
+
+    internal SentryFailedRequestHandler(IHub hub, SentryOptions options)
+    {
+        _hub = hub;
+        _options = options;
+    }
+
+    public void HandleResponse(HttpResponseMessage response)
+    {
+        // Ensure request is not null
+        if (response.RequestMessage is null)
+        {
+            return;
+        }
+
+        // Don't capture if the option is disabled
+        if (!_options.CaptureFailedRequests)
+        {
+            return;
+        }
+
+        // Don't capture events for successful requests
+        if (!_options.FailedRequestStatusCodes.Any(range => range.Contains(response.StatusCode)))
+        {
+            return;
+        }
+
+        // Ignore requests to the Sentry DSN
+        var uri = response.RequestMessage.RequestUri;
+        if (uri != null)
+        {
+            if (_options.Dsn is { } dsn && new Uri(dsn).Host.Equals(uri.Host, StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            // Ignore requests that don't match the FailedRequestTargets
+            var requestString = uri.ToString();
+            if (!_options.FailedRequestTargets.ContainsMatch(requestString))
+            {
+                return;
+            }
+        }
+
+#if NET5_0_OR_GREATER
+        // Starting with .NET 5, the content and headers are guaranteed to not be null.
+        var bodySize = response.Content.Headers.ContentLength;
+#else
+        // We have to get the content body size before calling EnsureSuccessStatusCode,
+        // because older implementations of EnsureSuccessStatusCode disposes the content.
+        // See https://github.com/dotnet/runtime/issues/24845
+
+        // The ContentLength might be null (but that's ok).
+        // See https://github.com/dotnet/runtime/issues/16162
+        var bodySize = response.Content?.Headers?.ContentLength;
```

Comment:
@mattjohnsonpint this was the approach I'd originally tried. However it's not that it returns null - when running tests against the net43 target it sometimes throws a System.ObjectDisposedException... so I don't think this code is safe. 

I also tried variants like `response.Content?.Headers.?.Contains("Content-Length")` as well, which avoided the exception on net43 but also prevented us from capturing the BodySize in later versions of .NET where it could be calculated but wasn't stored in the header. 

It looks like [HttpContent.TryComputeLength](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpcontent.trycomputelength?view=net-8.0) (which I think is where they fixed all this) was introduced in .NET 4.5... so possibly `NET5_0_OR_GREATER` is the wrong cut-off point... 

An alternative I toyed with was `try... catch(ObjectDisposedException ex) {}`. Possibly that's a better solution but it would definitely need to be accompanied by a code comment explaining why it was there... we'd need to keep using an `#if` directive in the unit tests, but it would mean we'd be capturing the BodySize in all circumstances where this was possible (failing only when an exception was thrown by older versions of the .NET Framework).

Reply:
It's safe now, as it's been moved before `EnsureSuccessStatusCode` which used to dispose the content (which is why you got `ObjectDisposedException`).  See my comment just above this one.

As for specifically checking `"Content-Length"` - we could add that if `bodySize == null`, but I'd want to reproduce the issue first.  I tried locally (on both NetFx and Mono) and the value was always present if it came from an actual `HttpClient` instance.  And if it came just from `ResponseMessage` (like in the tests), then `Headers` was null, so there'd be nothing to check against.

Either way, I'm not too worried about it.  The disposed exception is the main thing, which is taken care of.

Reply:
Ah, interesting... How did you work out where Dispose was being called? 

I also just reaslised that relying on EnsureSuccessStatusCode to throw an exception means it's not possible to configure `FailedRequestStatusCodes` in the 200-299 range... nothing in that range will result in an exception being thrown here that will be captured by Sentry. 

I can't think why someone would want to do that... Something to be aware of though.